File: packages/shake/ios/Plugin/Classes/Options/StartWatchingOptions.swift
import Foundation
import Capacitor

@objc public class StartWatchingOptions: NSObject {
    private static let sensitivityHard = "hard"
    private static let sensitivityLight = "light"
    private static let sensitivityMedium = "medium"

    private static let shakeThresholdHard = 3.4
    private static let shakeThresholdLight = 2.3
    private static let shakeThresholdMedium = 2.7

    let shakeThreshold: Double

    init(_ call: CAPPluginCall) throws {
        let sensitivity = call.getString("sensitivity", StartWatchingOptions.sensitivityMedium)
        self.shakeThreshold = try StartWatchingOptions.shakeThreshold(for: sensitivity)
    }

    private static func shakeThreshold(for sensitivity: String) throws -> Double {
        switch sensitivity {
        case sensitivityHard:
            return shakeThresholdHard
        case sensitivityLight:
            return shakeThresholdLight
        case sensitivityMedium:
            return shakeThresholdMedium
        default:
            throw CustomError.invalidSensitivity
        }
    }
}

File: packages/shake/android/src/main/java/io/capawesome/capacitorjs/plugins/shake/classes/options/StartWatchingOptions.java
package io.capawesome.capacitorjs.plugins.shake.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.shake.classes.CustomExceptions;

public class StartWatchingOptions {

    public static final String SENSITIVITY_HARD = "hard";
    public static final String SENSITIVITY_LIGHT = "light";
    public static final String SENSITIVITY_MEDIUM = "medium";

    private static final float SHAKE_THRESHOLD_HARD = 3.4f;
    private static final float SHAKE_THRESHOLD_LIGHT = 2.3f;
    private static final float SHAKE_THRESHOLD_MEDIUM = 2.7f;

    private final float shakeThreshold;

    public StartWatchingOptions(@NonNull PluginCall call) throws Exception {
        String sensitivity = call.getString("sensitivity", SENSITIVITY_MEDIUM);
        this.shakeThreshold = StartWatchingOptions.getShakeThresholdForSensitivity(sensitivity);
    }

    public float getShakeThreshold() {
        return shakeThreshold;
    }

    private static float getShakeThresholdForSensitivity(@NonNull String sensitivity) throws Exception {
        switch (sensitivity) {
            case SENSITIVITY_HARD:
                return SHAKE_THRESHOLD_HARD;
            case SENSITIVITY_LIGHT:
                return SHAKE_THRESHOLD_LIGHT;
            case SENSITIVITY_MEDIUM:
                return SHAKE_THRESHOLD_MEDIUM;
            default:
                throw CustomExceptions.INVALID_SENSITIVITY;
        }
    }
}
